crates/service/src/async_worker.rs

```diff
@@ -0,0 +1,64 @@
+//! Async service worker task.
+
+use futures::FutureExt;
+use tokio::sync::watch;
+use tracing::*;
+
+use super::*;
+
+/// Async worker task.
+pub(crate) async fn worker_task<S: AsyncService, I>(
+    mut state: S::State,
+    mut inp: I,
+    status_tx: watch::Sender<S::Status>,
+    shutdown_guard: strata_tasks::ShutdownGuard,
+) -> anyhow::Result<()>
+where
+    I: AsyncServiceInput<Msg = S::Msg>,
+{
+    let mut exit_fut = Box::pin(shutdown_guard.wait_for_shutdown().fuse());
+    let mut wkr_fut = Box::pin(worker_task_inner::<S, I>(&mut state, &mut inp, &status_tx).fuse());
+
+    futures::select! {
+        _ = exit_fut => (),
+        res = wkr_fut => res?,
+    };
+
+    Ok(())
+}
+
+async fn worker_task_inner<S: AsyncService, I>(
```

Comment:
We don't have a graceful shutdown for the `async_worker` like we do for the `sync_worker`: https://github.com/alpenlabs/alpen/blob/d33444305b753a5606a557969f3a1b252f1aac8e/crates/service/src/sync_worker.rs#L40-L44

This can be something straight out of `strata-tasks`'s `ShutdownGuard`: https://github.com/alpenlabs/alpen/blob/d33444305b753a5606a557969f3a1b252f1aac8e/crates/tasks/src/shutdown.rs#L54-L80
or adapted if needed.